scripts/perf/format-perf-report.py

```
@@ -22,167 +28,613 @@ class Dist:
+    )
+
+
+def load_history_runs(history_dir: Optional[Path]) -> list[dict[str, Any]]:
+    if history_dir is None or not history_dir.exists() or not history_dir.is_dir():
+        return []
+    runs: list[dict[str, Any]] = []
+    for path in sorted(history_dir.glob("*.json")):
+        try:
+            run = load_json(path)
+        except Exception:
+            continue
```

Ruff (0.15.1)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)